app/src/lib/stores/app-store.ts

```diff
@@ -4596,6 +4597,7 @@ export class AppStore extends TypedBaseStore<IAppState> {
         ourBranch: tip.branch.name,
         theirBranch: sourceBranch.name,
       })
+      this._endMultiCommitOperation(repository)
```

Comment:
These two `this._endMultiCommitOperation(repository)` lines make sure the flow ends when the merge finishes and there are no conflicts. If there are conflicts, the multi commit operation will be ended after the conflicts are resolved.